addons/website/models/website.py

```diff
@@ -261,10 +261,10 @@ def _handle_domain(self, vals):
             vals['domain'] = vals['domain'].rstrip('/')
 
     @api.ondelete(at_uninstall=False)
-    def _unlink_except_last_remaining_website(self):
-        website = self.search([('id', 'not in', self.ids)], limit=1)
-        if not website:
-            raise UserError(_('You must keep at least one website.'))
+    def _unlink_except_default_website(self):
```

Comment:
We can't change signature in stable (at least unless for extreme need without workaround, which is far from the case here).
Keep the same name, eventually with a `TODO: rename in master` and adapt in forward port if you really want to.